aten/src/ATen/native/mps/operations/RangeFactories.mm

```diff
@@ -13,6 +13,42 @@
 namespace at {
 namespace native {
 
+static const char* ARANGE_KERNEL = R"METAL(
+#include <metal_stdlib>
+
+using namespace metal;
+struct ARangeFloatParam {
+  float start;
+  float step;
+  uint length;
+};
+
+kernel void arange_float(constant ARangeFloatParam *params [[buffer(0)]],
```

Comment:
Is `float` the only type which is supported ? Do we need a templatized version which can support other types?